filters/tee/tee.go

```diff
@@ -160,6 +160,12 @@ func (r *tee) Request(fc filters.FilterContext) {
 		}()
 
 		rsp, err := r.client.Do(copyOfRequest)
+
+		// Prevent to leak goroutines when request body is never read.
+		if fc.Shunted() {
+			defer copyOfRequest.Body.Close()
```

Comment:
As far as I remember the pipe is written to, as soon the original body is read.
Therefore I think you should read the body, because of the pipe and you also want to execute the tee() call and stream the body, right?

Reply:
I think that path works fine. If you have a backend, then the req.body is read and closed properly. 
On the other hand, if you shunt the response, it means that you are not interested in the response. Hence, we can close it. 

Also, I am now thinking that the defer does not have so much sense as the intention is to close the request immediately. 

**To summarize the intention:**
- If the route is shunted then, as soon as we applied this filter, we close the request body. In other words, the tee filter triggers a side effect. The client receives a 404 but there is a request executed from skipper to the backend we are teeing.

- If the route is not shunted then the original request is read and pipes the body to the copy of the request. It means that goes through the Read method of teeTie which consumes the body, pipes the contents and closes the pipe. This is the most common flow.

Let me know if this does it make sense.

Reply:
I will close this PR as we discussed better ways to fix this with @aryszka.